subprojects/core-api/src/main/java/org/gradle/api/artifacts/DependencySubstitutions.java

```diff
@@ -102,9 +103,41 @@
          */
         Substitution because(String reason);
 
+        /**
+         * Specifies that the substituted target dependency should use the specified classifier.
+         *
+         * This method assumes that the target dependency is a jar (type jar, extension jar).
+         *
+         * @since 6.6
+         */
+        @Incubating
+        Substitution usingClassifier(String classifier);
+
+        /**
+         * Specifies that the substituted dependency mustn't have any classifier.
+         * It can be used whenever you need to substitute a dependency which uses a classifier into
+         * a dependency which doesn't.
+         *
+         * This method assumes that the target dependency is a jar (type jar, extension jar).
+         *
+         * @since 6.6
+         */
+        @Incubating
+        Substitution withoutClassifier();
+
         /**
          * Specify the target of the substitution.
          */
         void with(ComponentSelector notation);
+
+
+        /**
+         * Specify the target of the substitution. This is a replacement for the {@link #with(ComponentSelector)}
+         * method which supports chaining.
+         *
+         * @since 6.6
+         */
+        @Incubating
+        Substitution to(ComponentSelector notation);
```

Comment:
We discussed which term to use there. I like your proposal Jendrik. So 👍 from me.

Reply:
Neither of these sound like natural English to me. I would suggest:
- substitute X for Y ("Y gets replaced with X")
- substitute X with Y ("X gets replaced by Y")

Saying "substitute X by Y" is unambiguous ("X gets replaced by Y"), but it sounds strange to me. Why are we not sticking with "with"?

Reply:
This is just a binary compatibility issue. We already have a `with` method which returns `void` so we can't change its return type. We can't use `for` either because it's a reserved keyword. I was just tempted to choose `using`:

`substitute (...) using (...)`

Not as nice as `with` but we can't change the past!

Reply:
Went ahead and changed to `using`.